File: osp-be/src/main/java/it/fincons/osp/dto/RichiestaIntegrazioneDTO.java
package it.fincons.osp.dto;

import java.time.LocalDateTime;

import javax.validation.constraints.NotNull;
import javax.validation.constraints.Size;

import it.fincons.osp.model.TipoRichiestaIntegrazione;
import lombok.Data;

@Data
public class RichiestaIntegrazioneDTO {
	private Long id;

	@NotNull
	private Long idPratica;

	@NotNull
	private Long idUtenteRichiedente;

	private Integer idStatoPratica;

	@Size(max = 100)
	private String codiceProtocollo;
	
	private LocalDateTime dataProtocollo;

	private TipoRichiestaIntegrazione tipoRichiesta;

	@Size(max = 512)
	private String motivoRichiesta;

	private LocalDateTime dataInserimento;

	private LocalDateTime dataScadenza;
	
	private IntegrazioneDTO integrazione;

	private String flagEsenzioneMarcaDaBollo;

	private Long idAllegato;

}

File: osp-fe/src/app/shared/models/dto/richiesta-integrazione-dto.ts
import { IntegrazioneDTO } from "./integrazione-dto";

export class RichiestaIntegrazioneDTO {
	id: number;
	idPratica: number;
	idUtenteRichiedente: number;
	idStatoPratica: number;
	codiceProtocollo: string;
	tipopRichiesta: string;
	motivoRichiesta: string;
	dataInserimento: Date;
	dataScadenza: Date;
	integrazione: IntegrazioneDTO;
	flagEsenzioneMarcaDaBollo: boolean;
	idAllegato: number;	
}
